phoenix-core-client/src/main/java/org/apache/phoenix/iterate/ScanningResultIterator.java

```diff
@@ -85,6 +86,8 @@ public class ScanningResultIterator implements ResultIterator {
 
     private long dummyRowCounter = 0;
 
+    private AtomicBoolean hasFirstValidResult = new AtomicBoolean(false);
```

Comment:
Phoenix result iterators are single threaded. We do not need to use AtomicBoolean.